source/develop/release-management/features/network/sriov-live-migration.md

```diff
@@ -0,0 +1,178 @@
+to the VM, and the user space. It is achieved by `failover` netdev which acts
+as master device and controls two slave devices, the `primary` device being 
+SR-IOV passtrough and the virtio as `standby`. The VM user access network
+through the `failover` device. 
+
+There is still need to provide two interfaces, the SR-IOV passtrough device
+and virtio device, to the affected VM, but the  overall configuration overhead
+has been reduced to selecting suitable vNIC profile pair.
+
+
+## Prerequisites
+
+The implementation requires a [Linux kernel][5], [libvirt][6] and [qemu][7] 
+support with `net_failover`. 
+
+The [Linux kernel][5] support has to be only on guest side. 
+
+## Limitations
+
+* In addition to the SR-IOV NIC, a linux-bridge based (or an OVN physical network in OVS clusters)
+  has to be connected to the source, and the destination host, marked as required.
+
+## Implementation
+
+### Entity Description
+
+vNIC profile will get a new attribute that will point to the failover virtio 
+vNIC profile. This selection will be only available with `Passtrough` and `Migratable`
+selected. 
+
+### User Experience
+
+Upon creation of vNIC profile, the user can enable failover and select 
+the failover network vNIC profile. The `Failover` selection is available 
+only for `Passtroguh` and `Migratable` vNIC profiles.
+
+REST API takes an additional parameter for vNIC called `failover`
+containing id of the failover vNIC profile:
+
+```
+<vnic_profile id="123">
+  <name>vNIC_name</name>
+  <pass_through>
+    <mode>enabled</mode>
+  </pass_through>
+  <migratable>true</migratable>
+  <failover id="456"/>
+</vnic_profile>
+```
+
+VM interface then requires only single vNIC profile that has the `failover` attribute 
+in order to utilize this behavior.
+
+### Engine backend
+
+As engine is responsible for creating XML for a new VM nic, the 
+appropriate XML will be created and provided to Libvirt. 
+
+In order to ensure that the virtio network is available it has to be
+marked as required, and the migration policy needs to be updated.
+
+### Libvirt XML
+
+If the feature is activated for a new VM nic, the libvirt XML definition
+contains an additional network with attribute [teaming element][8],
+as follows:
+
+```
+<devices>
+  <interface type='network'>
+    <source network='mybridge'/>
+    <mac address='00:11:22:33:44:55'/>
+    <model type='virtio'/>
+    <teaming type='persistent'/>
+    <alias name='ua-backup0'/>
+  </interface>
+  <interface type='network'>
+    <source network='hostdev-pool'/>
+    <mac address='00:11:22:33:44:55'/>
+    <model type='virtio'/>
+    <teaming type='transient' persistent='ua-backup0'/>
+  </interface>
+</devices>
+```
+
+### Vdsm
+
+Handling of update and device removal is handled in vdsm. In both cases we 
+can actually apply changes that are relevant to both interface as those 
+interfaces are linked by the `teaming` attribute. 
+
+Relevant update changes are:
+* Nic unplug
+* MAC address change 
+
```

Comment:
@erav Can you please have a look? You might be deep in the details here.

Reply:
* Nic unplug:
Is the failover attribute immutable after the profile creation?
if not is toggling it via a vnic profile update applied immediately to the device or is a vnic replug required? this might have an implication to the out-of-sync indication.
* MAC update:
IIUC there is a single vNIC profile with a single MAC address for both devices. So from the point of view of engine's MAC pool infrastructure this setup should not be different than any other vnic profile. if there are duplicate addresses in the pool, the migration might run into collisions regardless of the failover feature.

Reply:
>     * Nic unplug:
>       Is the failover attribute immutable after the profile creation?
>       If not, is toggling it via a vnic profile update applied immediately to the device or is a vnic replug required? this might have an implication to the out-of-sync indication.

It will be editable, and does not have immediate effect. So this should be also part of the out-of-sync flag. 

> 
>     * MAC update:
>       IIUC there is a single vNIC profile with a single MAC address for both devices. So from the point of view of engine's MAC pool infrastructure this setup should not be different than any other vnic profile. if there are duplicate addresses assigned from the pool, the migration might run into collisions regardless of the failover feature.

Yes, the nic behaves the same in regards of MAC addresses only under the hood there are two nics with same MAC.